From 6e85202b1f98ac359153db13da5d0aff6bb6a52f Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Tue, 30 Aug 2016 15:37:43 +0200
Subject: [PATCH 110/110] SYSDB: Fix error handling in
 sysdb_get_user_members_recursively
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We ignored failures from sysdb_search_entry

Reviewed-by: Petr ÄŒech <pcech@redhat.com>
(cherry picked from commit b969ccc2cc58fdf761e5d314de9217f2d914bc9b)
---
 src/db/sysdb_ops.c   | 3 +++
 src/db/sysdb_views.c | 5 ++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/db/sysdb_ops.c b/src/db/sysdb_ops.c
index 8756ffee164335aea5d5aa6258ccf7406faee47e..a4aff0178cffc23da6cb5ce1e6d4599c4f2a18d9 100644
--- a/src/db/sysdb_ops.c
+++ b/src/db/sysdb_ops.c
@@ -3938,6 +3938,9 @@ errno_t sysdb_get_user_members_recursively(TALLOC_CTX *mem_ctx,
 
     ret = sysdb_search_entry(tmp_ctx, dom->sysdb, base_dn, LDB_SCOPE_SUBTREE,
                              filter, attrs, &count, &msgs);
+    if (ret != EOK) {
+        goto done;
+    }
 
     res = talloc_zero(tmp_ctx, struct ldb_result);
     if (res == NULL) {
diff --git a/src/db/sysdb_views.c b/src/db/sysdb_views.c
index 284f21f5b46ce4e27af0253752b00915048c0ef5..30e0c07ea8037e6f9af687930a3a921100c796fc 100644
--- a/src/db/sysdb_views.c
+++ b/src/db/sysdb_views.c
@@ -1299,7 +1299,10 @@ errno_t sysdb_add_group_member_overrides(struct sss_domain_info *domain,
 
     ret = sysdb_get_user_members_recursively(tmp_ctx, domain, obj->dn,
                                              &res_members);
-    if (ret != EOK) {
+    if (ret == ENOENT) {
+        ret = EOK;
+        goto done;
+    } else if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE,
               "sysdb_get_user_members_recursively failed.\n");
         goto done;
-- 
2.7.4

